tmexp/io_constants.py

```diff
@@ -81,6 +81,13 @@ class DocumentEvolution(NamedTuple):
 METRICS_FILENAME = "metrics.pkl"
 LABELS_FILENAME = "labels.txt"
 
+
+DeltaMapping = Dict[str, int]
+RefMapping = DefaultDict[str, DeltaMapping]
+DocMapping = DefaultDict[str, RefMapping]
+RepoMapping = DefaultDict[str, DocMapping]
```

Comment:
For all of those it's also possible to subclass defaultdict to avoid painful init:
```python
class DeltaMapping(Dict[str, int]):
    pass


class RefMapping(DefaultDict[str, DeltaMapping]):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.default_factory = DeltaMapping


class DocMapping(DefaultDict[str, RefMapping]):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.default_factory = RefMapping


class RepoMapping(DefaultDict[str, DocMapping]):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.default_factory = DocMapping
```

Reply:
nice !